linkerd/app/core/src/svc.rs

```diff
@@ -172,13 +179,19 @@ impl<S> Stack<S> {
         self.push(stack::OnServiceLayer::new(layer))
     }
 
-    /// Wraps the inner `S` with `NewCloneService` so that the stack holds a
+    /// Lifts the inner `S` with `NewCloneService` so that the stack holds a
     /// `NewService` that always returns a clone of `S` regardless of the target
     /// value.
-    pub fn push_new_clone(self) -> Stack<NewCloneService<S>> {
+    pub fn lift_new(self) -> Stack<NewCloneService<S>> {
         self.push(layer::mk(NewCloneService::from))
     }
 
+    /// Lifts the inner stack via [`Self::lift_new`], but combines both target
+    /// types into a new `P`-typed value via `From`.
+    pub fn lift_new_with_target<P>(self) -> Stack<NewFromTargets<P, NewCloneService<S>>> {
+        self.lift_new().push(NewFromTargets::layer())
+    }
```

Comment:
`lift_new` takes a `NewService<T>` and returns a `NewService<Z, Service = NewService<T>>` -- turning it into a double-NewService that ignores the first target.

`NewFromTargets` takes a `NewService<T>` and returns a `NewService<T, Service = NewServie<(U, T)>` -- so that first target is cloned and combined with the second target.

The result is that we expose `NewService<T, Service = NewService<U>>` over an inner `NewService<(U, T)>`

Reply:
yeah, i figured that out, i just felt like it might not be clear to other readers.